File: 6-ZigzagConversion/6-ZigzagConversion.cpp
// Last updated: 11/8/2025, 4:09:10 PM
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1) return s; 
        double cycles = s.length() / double(2 * numRows - 2);
        int numCycles = ceil(cycles);
        int numCols = numCycles * (numRows - 1);
         
         vector<string> myVec(numRows);

         int i = 0; // rows
         
        bool isVertical = true;
        int t = 0;
        while( t < s.length()){
            if(isVertical){
                while(i < numRows && t < s.length()){
                myVec[i]+=(s[t]);
                t++;
                i++;

                }
                isVertical = false;
                i -= 2;
            }
            else{
                for (int q = 0; q < numRows - 2 && t < s.length(); q++){
                    myVec[i]+=(s[t]);
                    t++;
                    i--;
                }
                isVertical = true;
            }

        }

            string toReturn;
        for (int n = 0; n < numRows; n++){
            toReturn += myVec[n];
        }

        return toReturn;
    }
};

File: 6-ZigzagConversion/6-ZigzagConversion.py
# Last updated: 11/8/2025, 8:22:37 PM
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s
        
        # cycles = len(s) / (2 * numRows - 2)
        # numCycles = math.ceil(cycles)
        # numCols = numCycles * (numRows - 1)
        
        myVec = [''] * numRows
        
        i = 0  # rows
        isVertical = True
        t = 0
        
        while t < len(s):
            if isVertical:
                while i < numRows and t < len(s):
                    myVec[i] += s[t]
                    t += 1
                    i += 1
                
                isVertical = False
                i -= 2
            else:
                for q in range(numRows - 2):
                    if t >= len(s):
                        break
                    myVec[i] += s[t]
                    t += 1
                    i -= 1
                
                isVertical = True
        
        return ''.join(myVec)

        